src/modules/site-v2/templates/primary/map.html

```diff
@@ -261,43 +264,40 @@
     }
   });
 
+  const slice_idx1 = elegans.length/3
+  const slice_idx2 = (elegans.length - slice_idx1)/2
+  const elegans1 = elegans.slice(0, slice_idx1)
+  const elegans2 = elegans.slice(slice_idx1, slice_idx2)
+  const elegans3 = elegans.slice(slice_idx2)
+  L.layerGroup(elegans1).addTo(map)
+  L.layerGroup(elegans2).addTo(map)
+  L.layerGroup(elegans3).addTo(map)
+
+
   const elegans_layer = L.layerGroup(elegans)
   const briggsae_layer = L.layerGroup(briggsae)
   const tropicalis_layer = L.layerGroup(tropicalis)
+  const layers_obj = {
+    'elegans_layer': elegans_layer,
+    'briggsae_layer': briggsae_layer,
+    'tropicalis_layer': tropicalis_layer
+  }
 
   let overlays = L.layerGroup().addLayer(elegans_layer).addTo(map)
 
-  function showStrains() {
-    const filters = document.getElementById('strains').filters
-
-    const checked_species = []
-    for (let i = 0; i < filters.length; i++) {
-      if(filters[i].checked) checked_species.push(filters[i].value)
-    }
-    overlays.clearLayers()
-
-    if(checked_species.includes('elegans_layer')) {
-    overlays.addLayer(elegans_layer)
-    }
-    if(checked_species.includes('briggsae_layer')) {
-    overlays.addLayer(briggsae_layer)
-    }
-    if(checked_species.includes('tropicalis_layer')) {
-    overlays.addLayer(tropicalis_layer)
-    }
-  }
-
   document.querySelector('#strains').addEventListener('click', e => {
     if (e.target.tagName !== 'INPUT') return;
-    
     const input = e.target
     const value = input.value;
-    showStrains(value)
+    const layer = layers_obj[value]
+    if (input.checked) {
+      overlays.addLayer(layer)
```

Comment:
will this add duplicate layers if someone clicks/unclicks/clicks the same species?